### ISSUE TITLE
Cloning dataframe with number column names reorders columns

### ISSUE THREAD

**Issue by wenyichen** [data-forge/data-forge-ts#101]
```
  const dataFrame1 = new DataFrame({
    columnNames: ['abc','123'],
    rows: [['a','1']]
  });
  console.log(dataFrame1.getColumnNames());
  const dataFrame2 = new DataFrame(dataFrame1);
  console.log(dataFrame2.getColumnNames())
```

```
['abc','123']
['123','abc']
```

**Comment by ashleydavis** [comment]
That's strange I haven't noticed that before. 

Are you able to dig deeper into the code to figure what's happening?

**Comment by wenyichen** [comment]
suspect it’s due to Object.keys behavior in this line but haven’t dived into it yet

https://github.com/data-forge/data-forge-ts/blob/b349e1beb3b970b653be90136cf420bd2b41ff68/src/lib/dataframe.ts#L2698

**Comment by ashleydavis** [comment]
That makes sense.

In this case the DataFrame you are passing into the constructor is treated as an iterable of objects and so the column order is not preserved.

It appears this is working as it should.

If you wanted to fix this you'd have to make the DataFrame constructor detect when it's passed another DataFrame, it could then initialize itself with the correct column order.

Feel free to propose a new constructor implementation that would work the way you want. You'd have to create unit tests as well, but in the first instance you could just get the code and hack it work the way you want then copy the updated code into this issue to show we what it would look like. We can discuss then talk about automated tests.

**Comment by ashleydavis** [comment]
This issue is fixed in Data-Forge v1.9.5.
